nucypher/policy/conditions/evm.py

```diff
@@ -95,9 +95,14 @@ def _resolve_any_context_variables(
         processed_parameters.append(p)
 
     v = return_value_test.value
+    k = return_value_test.key
```

Comment:
What's the scenario where the key for the test would be a context variable?

Presumably the `key` is determined at condition creation based on the specific ABI, and not at condition evaluation? But, maybe I'm missing something.

Reply:
if the function returned a struct mapping addresses to something, and we wanted to key by :userAddress

Reply:
It's plausible that struct can be returned that is keyed by addresses, which might be the user's address.  Whether or not that address needs to be recovered/verified by a digital signature is unclear from an authentication purposes, but it's certainly convenient UX opposed to having the user paste their own address into the condition.

Reply:
> Whether or not that address needs to be recovered/verified by a digital signature is unclear from an authentication purposes

^ With `:userAddress` it would need to be authenticated - that's the requirement of the processing directive for that context variable; the case for *needing* authentication for this case is not clear to me

Reply:
Consider that it might just be a matter of convenience since there is not an *unauthenticated* context variable for user address.

Reply:
i can't think of a case where the users address would need to be used without it being authenticated somehow.
Let's say i had a whitelisting contract with an attribute `whitelist` that mapped addresses to booleans.

I would want my return value test to look something like:
```
    "returnValueTest": {
      "key": :userAddress,
      "comparator": "==",
      "value": true
    }
```

I **need** the `:userAddress` to be authenticated here

Reply:
This is where my confusion lies 😅 , but perhaps there are other examples out there.

Perhaps it is possible that someone could write a contract that way, but seems unlikely that they would. To me, the case for whitelisting is a contract function that takes the `:userAddress` as a param and returns true/false i.e. `function isWhitelisted(address user) returns(bool)`; and it is as a param that the address would be authenticated, not as a key on the return value. Otherwise, the entire dictionary of who is whitelisted is returned, which:
1. could get long
2. reveals/leaks other addresses that are whitelisted

With the `isWhitelisted` function, the ReturnValueTest just checks for `true`/`false` and does not need the key. 

Basically what I'm wondering is, in any of the cases where some mapping keyed by address could be returned, would it be common to also have functions in the same contract that just take an address and return information specific to the address, since returning the entire internal mapping is not efficient.

Maybe @vzotova may have some thoughts here given her expertise in contracts, and common practices.

Reply:
I'm not sure we need to uncover every possible example.  Another way to approach this design decision: Do we want every possible input and output value to support `:userAddress` for maximum flexibility, or is less control still sufficient to cover practical use cases?

Reply:
> or is less control still sufficient to cover practical use cases?
That's along the lines I'm thinking.  The base case for a `key` to be specified is a great one. My thought here is to start simple (actual value), which could be sufficient to cover most practical use cases - and then iterate if needed. 

That being said, I accept if others feel differently.